### PR TITLE
[feature] Added server side logging

### LINKED ISSUE #82: [wifi-login-pages] Add server side logging (nodejs)
Add a proper logger to the server implementation.
Log HTTP requests to openwisp-radius with INFO.
When there is an error, ensure the logging works also if the request is canceled (eg: timeout). Right now in case of timeout the request logging fails and there's no logging.

### PR DESCRIPTION
**Pull request by codesankalp**
Closes #82

Tasks done:

- Added Winston for logging
- Integrated morgan with Winston to log HTTP requests
- Removed checking for status code to handle timeout error and connection refused error.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -325,6 +325,22 @@ login_form:
 
 In order to enable users to log via third-party services like Google and Facebook, the ["Social Login" feature of OpenWISP Radius](https://openwisp-radius.readthedocs.io/en/latest/user/social_login.html) must be configured and enabled.
 
+#### Configuring Logging
+
+There are certain environment variables used to configure server logging. The details of environment variables to configure logging are mentioned below:
+
+| Environment Variable | Detail                                                                                                                                                                                        |
+| -------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **LOG_LEVEL**        | (optional) This can be used to set the level of logging. The available values are `warn` and `debug`. By default log level is set to `debug` during development and `warn` during production. |
```
Comment (nemesisdesign) [inline README.md:334]: > The available values are `warn` and `debug`.

What about `info` and `error`?

File: server/loggerConfig.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const logFilePath = {
+  all: process.env.ALL_LOG_FILE || "logs/all.log",
+  error: process.env.ERROR_LOG_FILE || "logs/error.log",
+  warn: process.env.WARN_LOG_FILE || "logs/warn.log",
+  info: process.env.INFO_LOG_FILE || "logs/info.log",
+  http: process.env.HTTP_LOG_FILE || "logs/http.log",
+  debug: process.env.DEBUG_LOG_FILE || "logs/debug.log",
```
Comment (nemesisdesign) [inline server/loggerConfig.js:7]: Make sure that by default everything goes to one file.

**Review by nemesisdesign — CHANGES_REQUESTED** [review]

File: server/loggerConfig.js (hunk 2 above)
Comment (codesankalp) [reply server/loggerConfig.js:7]: By default, all the logs goes to `logs/all.log`

File: README.md (hunk 1 above)
Comment (codesankalp) [reply README.md:334]: Added

**Review by nemesisdesign — APPROVED** [review]

**Pull request merged**